src/fluent/watch.ts

```diff
@@ -198,6 +201,19 @@ export class Watcher<T extends GenericClass> {
     return this.#events;
   }
 
+  /**
+   * Read the serviceAccount Token
+   *
+   * @returns token or null
+   */
+  async #getToken() {
+    try {
+      return (await fs.promises.readFile(this.#TOKEN_PATH, "utf8")).trim();
```

Comment:
this feels odd for us to be calling on-demand against the filesystem, something I'd expect to do once and not each binding--also this whole wiring feels....odd

Reply:
Good point and agreed, this should be done only once, will correct that. 

The reason for this is an side-affect of not use `@kubernetes/client-node`  which does this [behind the scenes](https://github.com/kubernetes-client/javascript/blob/82c85b94290458385bf976ad9b801702953e11e1/src/config.ts#L455)

Reply:
The way to do this only once would be to move this read to Pepr